theano/sandbox/cuda/basic_ops.py

```diff
@@ -3340,6 +3318,65 @@ def c_code(self, node, name, inputs, out_, sub):
                 %(fail)s;
             }
         }
+        """ % locals()
+        return str
+
+    def infer_shape(self, node, input_shapes):
+        return [node.inputs]
+
+    def c_code_cache_version(self):
+        return (1,)
+
+    def do_constant_folding(self, node):
+        return False
+
+gpu_alloc_empty = GpuAllocEmpty()
+
+
+class GpuAlloc(GpuAllocEmpty):
+    """Implement Alloc on the gpu.
+
+    The memset_0 param is an optimization. When True, we call
+    cudaMemset that is faster.
+
+    """
+    __props__ = ('memset_0',)
+
+    def __init__(self, memset_0=False):
+        self.memset_0 = memset_0
+
+    def __str__(self):
+        #Hide the memset parameter when not used to prevent confusion.
+        if self.memset_0:
+            s = "%s{memset_0=%s}" % (self.__class__.__name__, self.memset_0)
+        else:
+            s = self.__class__.__name__
+        return s
+
+    def make_node(self, value, *shape):
+        #if there is unneeded transfert generated by the next line
+        #the optimizer will remove them.
+        v = as_cuda_ndarray_variable(value)
+        shape, output = self.validate_shape(shape)
+        return Apply(self, [v] + shape, [output])
+
+    def perform(self, node, inputs, out_):
+        # the super class (GpuAllocEmpty) allocates memory, we fill it
+        value = inputs[0]
+        shps = inputs[1:]
+        super(GpuAlloc, self).perform(node, shps, out_)
+        out, = out_
+        out[0][...] = value  # broadcast value to fill us up
+
+    def c_code(self, node, name, inputs, out_, sub):
+        # the super class (GpuAllocEmpty) allocates memory, we fill it
+        value = inputs[0]
+        shps = inputs[1:]
+        str = super(GpuAllocEmpty, self).c_code(node, name, shps, out_, sub)
```

Comment:
str is a python keyword. Try to not override it. Could you rename it to something else?